aspnetcore/blazor/Security-guidance.md

```diff
@@ -0,0 +1,254 @@
+  * It validates that there is an event handler associated with that event.
+
+For a XSS vulnerability to be possible, the application must allow some user input to be included as part of the rendered page. Blazor server-side components go through a compile-time step where the contents of the page are transformed into what we call a RenderTree (a virtual DOM like structure) and the diffs between previous and current renders are sent to the client for rendering, or in the case of prerendering serialized into HTML.
+* At this step blazor server-side takes care of HTML encoding all content that wasn't defined as static markup at compile-time.
+* Script tags aren't allowed to be included as components and a compile time error is generated when one such tag is included in a razor page.
+  * Script tags will not behave the way one would expect on the page. Once added they execute the contents of the script and those side-effects don't disappear even if you remove the script tag.
+* Component authors can author components in C# without using Razor. For these cases, the component author is responsible for encoding all potentiall user input from the component.
+* It is also recommended that you avoid including script tags as part of your component render tree for the reasons stated above.
+
+As part of protecting against XSS attacks we recommend looking into application specific mitigations like content security policy (CSP). See https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP for more details.
+
+## Cross-origin protection
+Cross-origin attacks involve a client from a different origin performing some action against your server. This action is typically a form POST (CSRF), but opening a websocket is also possible. Blazor server-side applications offer the same guarantees that any other SignalR application using the hub protocol offers.
+* A browser from a different origin can't connect to the blazor-server side application and start a new session (circuit) with the default application configuration.
+  * This is due to CORS being disabled by default and SignalR forcing a preflight request against the server as part of the hub protocol handshake.
+* If you enable CORS in your application then you might need to take extra steps to protect your application depending on your CORS configuration.
+  * If you enable CORS globally you can disable it specifically for the blazor server-side hub by adding the `DisableCorsAttribute` metadata to the endpoint metadata after calling `hub.MapBlazorHub()`.
+
+## Click-jacking
+Click-jacking involves rendering a site as an iframe inside a site from a different origin in order to trick the user into performing actions on the site under attack.
+
+If you want to protect your application from being rendered inside an iframe we recommend using CSP and the `X-Frame-Options` header. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options for more details
```

Comment:
This can perhaps be replaced with `CSP frame-ancestors`. I think one problem perhaps with CSP is that the next step as to how and which features to implement isn't clear. One source for more information is the excellent https://scotthelme.co.uk/security-headers-updates/ (of @ScottHelme).